dxr/plugins/descriptor/__init__.py

```diff
@@ -119,9 +119,9 @@ def generic_describe(self):
         try:
             description_re = re.compile(r'(?:{}|{}|description)\
                                           (?:{})?\s*(?:[{}]\n?)\s*\
-                                          (?P<description>[\w\s-]+)'.format(re.escape(self.path),
-                                                                            re.escape(root),
-                                                                            re.escape(ext),
+                                          (?P<description>[\w\s-]+)'.format(re.escape(self.path.encode('utf-8')),
```

Comment:
Why go through special trouble to encode everything back to strs? Why not let the regex itself be unicode? Then it will actually match any applicable unicode pathnames from the unicode `self.contents`.

Witness:
```python
>>> re.compile(u'füd'.encode('utf-8')).match(u'füd') is None
True
>>> re.compile(u'füd').match(u'füd')
<_sre.SRE_Match object at 0x107bee510>
```